source/agora/utils/Test.d

```diff
@@ -624,62 +627,274 @@ unittest
+            assert(0, "Amount overflow");
+        auto remainder = amount.div(toward.length);
+
+        this.data.outputs = null;
+        foreach (key; toward)
+            this.data.outputs ~= Output(amount, key);
+
+        this.data.outputs[0].value.mustAdd(remainder);
+    }
+
+
+    /***************************************************************************
+
+        Distributes the attached input into multiple outputs with
+        the same amount.
+
+        Params:
+            share = amount to be distributed to one address
+            use_remainder = whether to use the remaining amount
+                after distribution.
+            toward = address of destination to be transferred.
+
+    ***************************************************************************/
+
+    public void distribute (scope const Amount share, bool use_remainder,
+        scope const PublicKey[] toward...) @safe
+    {
+        if ((this.inputs.length == 0) || (toward.length == 0))
+            return;
+
+        Amount sum;
+        if (!this.inputs.all!(val => sum.add(val.output.value)))
+            assert(0, "Amount overflow");
+        assert(sum >= share);
+
+        this.data.outputs = null;
+        foreach (key; toward)
+        {
+            this.data.outputs ~= Output(share, key);
+            assert(sum.sub(share), "Amount underflow");
+        }
+
+        if (use_remainder)
+        {
+            assert(this.inputs.length > 0);
+            if (sum > Amount(0))
+                this.data.outputs ~= Output(sum, this.inputs[0].output.address);
+        }
     }
-    return result;
+
+    /***************************************************************************
+
+        Merges the attached input into one output.
+
+        Params:
+            toward = address of destination to be transferred.
+
+    ***************************************************************************/
+
+    public void merge (scope const PublicKey toward) @safe
+    {
+        if (this.inputs.length == 0)
+            return;
+
+        Amount amount;
+        if (!this.inputs.all!(val => amount.add(val.output.value)))
+            assert(0, "Amount overflow");
+
+        this.data.outputs = null;
+        this.data.outputs ~= Output(amount, toward);
+    }
+
+    /***************************************************************************
+
+        Attaches one output of a transaction
+
+    ***************************************************************************/
+
+    public void attach (scope const Transaction tx, uint index) @trusted
+    {
+        this.inputs ~= cast(OutputRef)oref(tx, index);
+    }
+
+    /***************************************************************************
+
+        Attaches all outputs of a transaction
+
+    ***************************************************************************/
+
+    public void attach (scope const Transaction tx) @trusted
+    {
+        this.inputs ~= cast(OutputRef[])oref(tx).array;
```

Comment:
Why the `cast` ?

Reply:
The following message came out.
cannot append type const(OutputRef) to type OutputRef[]

Reply:
I'll change the member like this and remove `cast`.
``` D
private const(OutputRef)[] inputs;
```